### PR TITLE
Admin Dashboard: UX Responsive Enhancements

### PR DESCRIPTION
**Pull request by thabaum**
Updates the dashboard and UX

#448 redone

### CODE REVIEW

**Comment by thabaum** [comment]
This PR has a few tweaks in it.

Goal was to make the UX for the dashboard a more modern layout that gives a device friendly experience.

- More DIV and less Table tags used for UX
- Bootstrap styled classes added to Dashboard components
- Responsive tables
- Fixes Pane UI title class div container (now is 100% sized to pane)
- Other mods related to layout

This is a good starting point if nothing else I believe, let me know what you think I can polish it out further. 
Thank you.

**Comment by sbwalker** [comment]
Hi Cody. I really appreciate your efforts on this item. With less than a week until the 1.0 release I am not comfortable merging such a massive set of changes as there is not enough time for review or regression testing. I mentioned this on the other PR which you closed, but I think the best way to move forward is to break this down into smaller pieces which we can review, discuss, and begin to integrate after 1.0. My suggestion was to initially reduce the scope to a few modules, submit a PR, allow people to review and provide feedback, and once an approach is agreed upon make the changes to the remaining modules. In fact it would even be preferable if separate PRs were submitted for each module. Massive PRs are very difficult to review and merge.

**Comment by thabaum** [comment]
well i figured this would all get re-written down the road... this was just to make things "device friendly".  I have gone through and tested this a lot and I understand things can go differently.  This just makes things work.  And if you go through each page after firing this branch up as it is you will see everything works normally.  

This is my "massive" pr and a few things can get broken out of this into separate PR's that should be fixed in version 1.  

I will break those down to separate PR's if needed.  

If anyone interested can fire this up and find anything that could get dialed in further for just a quick.

I didn't play with any "backend" other than add a class that was missing from the Pane file to allow the full width of div to be used.

**Comment by sbwalker** [comment]
I will download your branch and do some local testing. I would also post a note in the Gitter channel for others to do the same. I am certain that we will be able utilize your changes - I am just not sure on the timing as we are so close to release. I trust this makes sense.

**Comment by thabaum** [comment]
as far as each page goes... there are issues all over here man.  For example the event log will show 10 events but is missing the pager to go to the next page.

We can go through each page and how it can be more responsive down the road.  I appreciate a crash test run :)

**Comment by thabaum** [comment]
couple of known issues:

- Tenant Management does not have "add" feature (probably not ready or not allowed on the installation?)
- Log viewer does not include pagination  (maybe we can include this now?)

**Comment by thabaum** [comment]
I see it should be using the pager, I am not sure if I have 10 logs yet but I figured the pager would be displayed maybe I need 11 logs?

File: Oqtane.Client/Modules/Admin/Pages/Index.razor [hunk 1]
```diff
@@ -3,23 +3,24 @@
...
-            <th>Name</th>
-        </Header>
-        <Row>
-            <td><ActionLink Action="Edit" Parameters="@($"id=" + context.PageId.ToString())" /></td>
-            <td><ActionDialog Header="Delete Page" Message="@("Are You Sure You Wish To Delete The " + context.Name + " Page?")" Action="Delete" Security="SecurityAccessLevel.Admin" Class="btn btn-danger" OnClick="@(async () => await DeletePage(context))" /></td>
-            <td>@(new string('-', context.Level * 2))@(context.Name)</td>
-        </Row>
-    </Pager>
-}
+<div class="container">
```
Comment (thabaum) [inline Oqtane.Client/Modules/Admin/Pages/Index.razor:6]: app css class selector could be included here and possibly even one in the pane div which loads prior to this one.  this has really no effect on the UX currently however was a work around for getting max width for the content pane.

**Comment by sbwalker** [comment]
@thabaum is it possible to submit PRs or at least identify any items you resolved in this PR which could be classified as "issues" in the current code? I am not talking about migrating from tables to divs - this only affects mobile scenarios and is a v.Next item. I am talking about anything which prevents functionality from working properly in a mainline desktop scenario. For example you mentioned an issue in Panes and there may be others.

**Comment by thabaum** [comment]
I submitted the PR mentioned.   The other will be similar to the prior one accepted however I will do up all pages in one PR tonight for that.

I will look through it for other smaller fixes as well.

> 
>     * More DIV and less Table tags used for dashboard components UX
> 
>     * Bootstrap styled classes added to Dashboard components
> 
>     * Responsive tables
> 
>     * Fixes Pane UI title class div container (now is 100% sized to pane)
> 
>     * Other mods related to layout
> 
>     * Permissions Grid and Pager adjustments
> 
>     * Changes disabled input to readonly fix to allow copy/paste but not allow changing the input field displayed. ****
>

**Comment by sbwalker** [comment]
I think you possibly misunderstand my question. I was asking about "anything which prevents functionality from working properly in a mainline desktop scenario". Most of the items you mentioned above do not fit this category and can be integrated in v.Next.

**Comment by thabaum** [comment]
the last two PR's you should add as they make the UI nicer I think.

This along with some smaller adjustments page by page like you said is fine.   I can see a few pages I would like to make adjustments to.

Two issues I can see with managing octane:

- Event Logs Pagination has an issue.  It will only show pagination when rows set to 50 or 100 but rows stay 10 and the pagination stays for 10 rows.  I was going to do an issue on this.  
-  Tenant management you cannot add a tenant.  However you can add a tennant if you add a site.

**Comment by thabaum** [comment]
image
this was another issue this PR resolves however I see some pages that can utilize more screen such as logs in my PR.

**Comment by thabaum** [comment]
What I would like to do post 1.0 release is discuss more about the css classes to be used.  

Like discussed in #442 creating a naming scheme for classes which I think we all have a similar approach to in mind here.  

app-admin-log-table
app-admin-log-form
app-admin-log-btn-group
app-admin-log-pagination

on the log page items.  Maybe we can build on that conversation in the weeks ahead and we can begin discussing each page.

I can see things getting chewed up here one page and component at a time.  Since this is not a v1 item it can marinate for some time here in this PR and submit single PR's as things look acceptable.

Without these classes it is harder to style as I removed all styles from forms here so they all are done with CSS classes as needed in future and can be modified easily down the road.

**Comment by thabaum** [comment]
And as far as using desktop nothing here resolves issues related to using desktop, only mobile.  Other than PR's already submitted that do relate to desktop have been committed to master.

The goal of this PR was to make the dashboard device friendly utilizing bootstrap.   The two PR's came from those efforts so not all lost here.  Everything is a win here as I am learning not only Blazor .NET Core and Visual Studio with GitHub but how to help with the project here and projects like this.  I am willing to help Bootstrap and Blazor projects to help Oqtane.  It is all Greek to me :)

**Comment by thabaum** [comment]
I will start chewing on this PR once the notifications PR's are merged.

We can discuss each component device responsiveness, layout and UX along with naming style classes to help allow styling each component if desired individually and give an example of how to create classes and style a component within the framework.

**Comment by thabaum** [comment]
So....... trying to recap this one.

No classes added but would like to add app-logs, app-notifications, app-profile and so on for each component section and sub sections.  We can do this one section at a time (PR) or you can make the changes as you like if you like and be done with it.

- Bootstrap Responsive Dashboard (expands full desktop and responds as needed similar to difference between login and registration ***which i did not touch if i recall correctly *** and other components)
- Changes some form input to custom-select if not folder related  (yes/no)

**Comment by hishamco** [comment]
@thabaum please resolve the conflicts and provide some screenshots

**Pull request closed without merging**